packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -186,6 +175,61 @@ describe("SharedString interval collections", () => {
 			]);
 		});
 
+		describe("changing endpoints and/or properties", () => {
+			it("changes only endpoints with new signature", () => {
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "hello world");
+				const id = collection.add({ start: 0, end: 3, props: { a: 1 } }).getIntervalId();
+
+				collection.change(id, { start: 1, end: 4 });
+
+				assertIntervalEquals(sharedString, collection.getIntervalById(id), {
+					start: 1,
+					end: 4,
+				});
+				assert.equal(collection.getIntervalById(id)?.properties.a, 1);
+			});
+			it("changes only properties with new signature", () => {
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "hello world");
+				const id = collection.add({ start: 0, end: 3, props: { a: 1 } }).getIntervalId();
+
+				collection.change(id, { props: { a: 2 } });
+
+				assertIntervalEquals(sharedString, collection.getIntervalById(id), {
+					start: 0,
+					end: 3,
+				});
+				assert.equal(collection.getIntervalById(id)?.properties.a, 2);
+			});
+			it("changes endpoints and properties with new signature", () => {
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "hello world");
+				const id = collection.add({ start: 0, end: 3, props: { a: 1 } }).getIntervalId();
+
+				collection.change(id, { start: 1, end: 4, props: { a: 2 } });
+
+				assertIntervalEquals(sharedString, collection.getIntervalById(id), {
+					start: 1,
+					end: 4,
+				});
+				assert.equal(collection.getIntervalById(id)?.properties.a, 2);
+			});
+			it("passes empty property set to change", () => {
+				const collection = sharedString.getIntervalCollection("test");
+				sharedString.insertText(0, "hello world");
+				const id = collection.add({ start: 0, end: 3, props: { a: 1 } }).getIntervalId();
+
+				collection.change(id, { props: {} });
+
+				assertIntervalEquals(sharedString, collection.getIntervalById(id), {
+					start: 0,
+					end: 3,
+				});
+				assert.equal(collection.getIntervalById(id)?.properties.a, 1);
+			});
+		});
```

Comment:
Add a test where both endpoints and properties are undefined since that is legal and should do nothing